packages/nameguard-react/src/components/ReportBadge/index.tsx

```diff
@@ -52,13 +62,36 @@ export function ReportBadge({
     "flex-shrink-0 appearance-none bg-white transition-colors hover:bg-gray-50 border border-gray-200 rounded-md px-2.5 py-1.5 inline-flex items-center";
   const buttonAndCursorClass = cc([buttonClass, "cursor-pointer"]);
 
-  const onClickHandler = () => {
-    if (onClickOverride) onClickOverride(ensName);
-    else {
-      window.location.href = getNameGuardURLForENSname(ensName);
+  const onClickHandler = (clickHandlerFor: ClickHandlerFor) => {
+    switch (clickHandlerFor) {
+      case ClickHandlerFor.badge:
+        if (onBadgeClickOverride) {
+          onBadgeClickOverride(ensName);
+        } else {
+          redirectUserToNameGuardInspectPage(ensName);
+        }
+        break;
+      case ClickHandlerFor.icon:
+        if (onIconClickOverride) {
+          onIconClickOverride(ensName);
+        } else {
+          redirectUserToNameGuardInspectPage(ensName);
+        }
+        break;
+      case ClickHandlerFor.tooltip:
+        if (onTooltipClickOverride) {
+          onTooltipClickOverride(ensName);
+        } else {
+          redirectUserToNameGuardInspectPage(ensName);
+        }
+        break;
     }
   };
 
+  const redirectUserToNameGuardInspectPage = (ensName: ENSName) => {
```

Comment:
Seems we should move this to `/utils/url.ts` ? Please see my other comment with more details about that proposed file.